applications/ThermalDEMApplication/custom_strategies/strategies/thermal_explicit_solver_strategy.cpp

```diff
@@ -73,6 +88,50 @@ namespace Kratos {
     KRATOS_CATCH("")
   }
 
+  //------------------------------------------------------------------------------------------------------------
+  void ThermalExplicitSolverStrategy::SetSolveFrequency(void) {
+    KRATOS_TRY
+    
+    // Parameters
+    ProcessInfo& r_process_info = GetModelPart().GetProcessInfo();
+    const double safety_factor  = 0.1;
+    const int    max_freq       = 10000;
+
+    // Get minimum diffusivity time between all particles
+    const int number_of_particles = (int)mListOfSphericParticles.size();
+    double min_diff_time = DBL_MAX;
+
+    #pragma omp parallel for
+    for (int i = 0; i < number_of_particles; i++) {
+      ThermalSphericParticle* particle = dynamic_cast<ThermalSphericParticle*>(mListOfSphericParticles[i]);
```

Comment:
besides, dynamic cast are quite tricky... @maceligueta what do you think?

Reply:
It is not recommended in general, but here we are using a FEM code for particles, and it is a quite an ‘incompressible’ FEM code. So, if you find a fast way to do “GetParticleDiffusivity()” which does not involve a search, a find, etc, we can dodge this downcasting and many others in DEM. I hate downcastings, but I hate even more GetValue(VARIABLE) or Calculate(VARIABLE), always in terms of performance, I mean.

Reply:
Fair enough :)